test/modelmanagers/articlemodel.test.js

```diff
@@ -0,0 +1,84 @@
+const { expect } = require('chai');
+const user = require('../../database/models').User;
+const { userprofile } = require('../../database/models');
+const article = require('../../lib/modelManagers/articlemodel');
+const constants = require('../mockData');
+
+
+let id;
+let res;
+let articleid, article2id;
+let testArticle, testArticle2;
+describe('Unit test article model manager functions', () => {
+  before(async () => {
+    res = await user.create(constants.userdata).then(async (newUser) => {
+      await userprofile.create(constants.userprofile);
+      return newUser;
+    });
+    ({ id } = res.dataValues);
+    testArticle = constants.article(id);
+    testArticle2 = constants.article(id);
+    article2id = testArticle2.id;
+  });
+
+  it('should create a new article when all necesary fields are provided', async () => {
+    const res = await article.createArticle(testArticle);
+    articleid = res.dataValues.id;
+    expect(res).to.be.an('object');
+    expect(res.dataValues.isPublished).to.be.equals(false);
+    expect(res.dataValues.title).to.be.equals(testArticle.title);
+    expect(res.dataValues.description).to.be.equals(testArticle.description);
+    expect(res.dataValues.content).to.be.equals(testArticle.content);
+    expect(res.dataValues.featuredImageUrl).to.be.equals(testArticle.featuredImageUrl);
+    expect(res.dataValues.averageRating).to.be.equals(testArticle.averageRating);
+    expect(res.dataValues.authorId).to.be.equals(testArticle.authorId);
+    expect(res.dataValues.slug).to.be.equals(testArticle.Slug);
+    expect(res.dataValues.readTime).to.be.equals(testArticle.readTime);
+  });
+
```

Comment:
Try `expect(res.dataValues).to.eql(testArtcles)` to compare objects. You may need to destructure your dataValues object. Check [here](https://medium.com/building-ibotta/testing-arrays-and-objects-with-chai-js-4b372310fe6d) for more.

Reply:
thanks for the resource, I have been able to implement this.